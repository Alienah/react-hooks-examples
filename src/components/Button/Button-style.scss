@import './../../styles/_vars.scss';

.button {
  * {
    border-style: solid;
    border-width: 1px;
    text-transform: uppercase;
  }
  .primary,
  .secondary {
    cursor: pointer;
    font-weight: bold;
  }

  .primary {
    border-color: $yellow1;
    &:hover {
      color: $white;
      background-color: $gray1;
      border-color: $white;
    }
    &:active {
      color: $gray1;
      background-color: $white;
      border-color: $gray1;
    }
  }
  .disabled {
    color: rgba(150, 149, 149, 0.70);
    background-color: rgba(0, 0, 0, 0.2);
    border-color: rgba(150, 149, 149, 0.70);

  }

  .secondary {
    background-color: $white;
    border-color: $gray1;
    color: $gray1;
    &:hover {
      -webkit-box-shadow: 0px 0px 8px 0px rgba(0,0,0,0.75);
      -moz-box-shadow: 0px 0px 8px 0px rgba(0,0,0,0.75);
      box-shadow: 0px 0px 8px 0px rgba(0,0,0,0.75);
      border-color: transparent;
    }
  }
}