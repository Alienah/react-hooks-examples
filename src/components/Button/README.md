# Default Button

```js
<Button>Click me</Button>
```

# Primary Button

```js
<Button type="primary">Click me</Button>
```

# Secondary Button

```js
<Button type="secondary">Click me</Button>
```

# Disabled Button

```js
<Button type="disabled">Click me</Button>
```