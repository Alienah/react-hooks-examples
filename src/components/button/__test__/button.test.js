import React from 'react';
//Para renderizar el dom en el test
import ReactDOM from 'react-dom';
import Button from './../button';

//Otra render diferente, pues es otra librería
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

//Snapshot
import ReactTestRenderer from 'react-test-renderer';

afterEach(cleanup);

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Button></Button>, div);
})

it('renders button correctly', () => {
  // Este es el render de la librería react testing library y devuelve varias cosas, una es el método getByTestId
  const {getByTestId} = render(<Button label="click me"></Button>)
  expect(getByTestId('button')).toHaveTextContent('click me');;
})

it('renders button correctly', () => {
  // Otro test con otro texto
  const {getByTestId} = render(<Button label="save"></Button>)
  expect(getByTestId('button')).toHaveTextContent('save');;
})

// Snapshop testing
it('matches snapshop', () => {
  // Esto creará un objeto. primero crea en el dom y luego lo transformamos a JSON
  const tree = ReactTestRenderer.create(<Button label="save"></Button>).toJSON();
  expect(tree).toMatchSnapshot();
})

it('matches snapshop with another text', () => {
  const tree = ReactTestRenderer.create(<Button label="click"></Button>).toJSON();
  expect(tree).toMatchSnapshot();
})