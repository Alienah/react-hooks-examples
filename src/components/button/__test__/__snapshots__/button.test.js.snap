// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshop 1`] = `
<div
  className="button"
  data-testid="button"
>
  save
</div>
`;

exports[`matches snapshop with another text 1`] = `
<div
  className="button"
  data-testid="button"
>
  click
</div>
`;
